indico/modules/events/ical.py

```diff
@@ -12,72 +12,121 @@
 
 from indico.core import signals
 from indico.core.config import config
+from indico.core.db.sqlalchemy.protection import ProtectionMode
 from indico.util.date_time import now_utc
 from indico.util.signals import values_from_signal
 
 
-def event_to_ical(event, user=None):
+def generate_basic_component(entity, uid=None, url=None):
+    """Generate an iCalendar component with basic common properties.
+
+    :param entity: Event/session/contribution where properties come from
+    :param uid: UID for the component
+    :param url: URL for the component (defaults to `entity.external_url`)
+
+    :return: iCalendar event with basic properties
+    """
+    component = Event()
+
+    component.add('dtstamp', now_utc(False))
+    component.add('dtstart', entity.start_dt)
+    component.add('dtend', entity.end_dt)
+    component.add('summary', entity.title)
+
+    if uid:
+        component.add('uid', uid)
+
+    if url:
+        component.add('url', url)
+    elif hasattr(entity, 'external_url'):
+        component.add('url', entity.external_url)
+
+    location = (f'{entity.room_name} ({entity.venue_name})'
+                if entity.venue_name and entity.room_name
+                else (entity.venue_name or entity.room_name))
+    if location:
+        component.add('location', location)
+
+    speaker_list = getattr(entity, 'person_links', [])
+    description = []
+    if speaker_list:
+        speakers = [f'{x.full_name} ({x.affiliation})' if x.affiliation else x.full_name
+                    for x in speaker_list]
+        description.append('Speakers: {}'.format(', '.join(speakers)))
+    if entity.description:
+        desc_text = str(entity.description) or '<p/>'  # get rid of RichMarkup
+        try:
+            description.append(str(html.fromstring(desc_text).text_content()))
+        except ParserError:
+            # this happens if desc_text only contains a html comment
+            pass
+    if len(description):
+        component.add('description', '\n'.join(description))
+
+    return component
+
+
+def generate_event_component(event):
+    """Generates an event icalendar component from an Indico event."""
+    uid = f'indico-event-{event.id}@{url_parse(config.BASE_URL).host}'
+    component = generate_basic_component(event, uid)
+
+    # add contact title, phones and emails if present
+    contact_info = [event.contact_title]
+    if event.contact_title:
+        if len(event.contact_emails):
+            contact_info += event.contact_emails
+        if len(event.contact_phones):
+            contact_info += event.contact_phones
```

Comment:
```suggestion
        contact_info += event.contact_emails
        contact_info += event.contact_phones
```

appending an empty list is a noop, so no need to check if it's empty first